typha/pkg/config/param_types.go

```diff
@@ -86,13 +86,14 @@ func (p *IntParam) Parse(raw string) (interface{}, error) {
 		err = p.parseFailed(raw, "invalid int")
 		return nil, err
 	}
-	result := int(value)
-	if result < p.Min {
+	if value < p.Min {
 		err = p.parseFailed(raw,
 			fmt.Sprintf("value must be at least %v", p.Min))
-	} else if result > p.Max {
+	} else if value > p.Max {
 		err = p.parseFailed(raw,
 			fmt.Sprintf("value must be at most %v", p.Max))
+	} else {
+		result := int(value)
```

Comment:
Think using `Atoi` above might be even better; it's defined to return `int` so it'll return something compatible with Min/Max or fail if the number doesn't fit in an int.